packages/evm/contracts/budgets/TransparentBudget.sol

```diff
@@ -145,6 +148,21 @@ contract TransparentBudget is ATransparentBudget, ReentrancyGuard {
         revert BoostError.NotImplemented();
     }
 
+    function clawbackFromTarget(address target, bytes calldata data_, uint256 boostId, uint256 incentiveId)
+        external
+        virtual
+        override
+        returns (uint256, address)
+    {
+        BoostLib.Boost memory boost = BoostCore(target).getBoost(boostId);
+        if (msg.sender != boost.owner) revert BoostError.Unauthorized();
```

Comment:
Nice this is a clever auth solution I was thinking we'd need some internal data to track this, this works much better.